ping.go

```diff
@@ -734,14 +736,17 @@ func (p *Pinger) sendICMP(conn packetConn) error {
 	}
 
 	for {
-		if _, err := conn.WriteTo(msgBytes, dst); err != nil {
+		_, err := conn.WriteTo(msgBytes, dst)
+		if err != nil {
 			if neterr, ok := err.(*net.OpError); ok {
 				if neterr.Err == syscall.ENOBUFS {
 					continue
 				}
 			}
+			p.PacketsSent++
 			return err
 		}
+
 		handler := p.OnSend
```

Comment:
I wonder if we also need to deal with the `p.OnSend` as well.

Reply:
An interesting point...

In my change I only wanted to fix the problem with the endlessly running ping that will not stop anymore.
My change now includes also the tried sends in the count of PacketsSent and not only the successful ones as before. 
And as I do not know what the original intention of the p.OnSend was and what it is already used for, I cannot provide any meaningful opinion to that.

To keep it simpler I would see the topic "should p.OnSend be only executed for successful sends" independent of the goal of my fix and worth to create a new, separate issue to discuss it ;)

Reply:
Looking this over again, I think this should be a separate counter, rather than increasing sent, since we didn't actually send anything

Reply:
I fully agree; I was struggling myself to go the simple path with just increasing PacketsSent because I feared the code would get more complicated. But I will give it a try and will check how it could look like.